File: packages/core/src/domain/domain-config.ts
import type { BaseAgent } from '../agent/base-agent';

export interface AgentDef {
  id: string;
  displayName: string;
  createAgent: () => BaseAgent;
  config?: Record<string, unknown>;
}

export interface EvalCriterion {
  name: string;
  weight: number;
  description: string;
}

export interface EvalRubric {
  criteria: EvalCriterion[];
}

/**
 * CSS custom-property name/value pairs only (CLAUDE.md: "theme: ThemeVars //
 * CSS custom properties only") — no component overrides or non-CSS hooks.
 */
export type ThemeVars = {
  [key: `--${string}`]: string;
};

export interface Rule {
  condition: string;
  threshold?: number;
  action: 'escalate_to_human' | 'flag_for_review' | 'notify_instructor';
}

/**
 * The ONLY place domain-specific values live (CLAUDE.md architecture rule).
 * NEXUS, packages/ui, and apps/* must never branch on domain identity —
 * everything domain-specific flows through this schema.
 */
export interface DomainConfig {
  id: string;
  name: string;
  subdomain: string;
  agents: AgentDef[];
  contentIndex: string;
  evalRubric: EvalRubric;
  theme: ThemeVars;
  escalationRules: Rule[];
}

File: apps/backend/domains/_contracts/domain_config.py
"""
Python mirror of packages/core/src/domain/domain-config.ts (TypeScript).
These are in-process configuration objects NEXUS loads at boot -- not
wire-format data crossing an HTTP boundary (unlike AgentInput/AgentOutput,
which are Pydantic) -- so they're plain dataclasses.
"""

from dataclasses import dataclass, field
from typing import Callable, Optional

from domains._contracts.base_agent import BaseAgent


@dataclass
class AgentDef:
    id: str
    display_name: str
    create_agent: Callable[[], BaseAgent]
    config: Optional[dict] = None


@dataclass
class EvalCriterion:
    name: str
    weight: float
    description: str


@dataclass
class EvalRubric:
    criteria: list[EvalCriterion] = field(default_factory=list)


@dataclass
class Rule:
    condition: str
    action: str
    threshold: Optional[float] = None


@dataclass
class DomainConfig:
    id: str
    name: str
    subdomain: str
    agents: list[AgentDef]
    content_index: str
    eval_rubric: EvalRubric
    theme: dict[str, str]  # CSS custom property name/value pairs
    escalation_rules: list[Rule]
